sound/soc/sof/intel/hda-dai.c

```diff
@@ -287,6 +287,37 @@ static int hda_link_pcm_prepare(struct snd_pcm_substream *substream,
 				  dai);
 }
 
+static int hda_link_dai_config_pause_push_ipc(struct snd_soc_dapm_widget *w)
```

Comment:
Isn't the fundamental reason to (have and) send SOF_DAI_CONFIG_FLAGS_2_STEP_STOP_PAUSE, to gracefully handle old FW versions? As this is the only way the hw works correctly, we'd otherwise just fix the code in both SW and FW. But as we have old FW shipping, we have to put this under a flag.

Reply:
hmm no the fundamental reason to send the IPC is to do the DMA pause in the FW after pipeline pause. Older FW will ignore this IPC and the new flag and take no action.

Reply:
@ranj063, and Linux will know how that the firmware is going to ignore this request?

Reply:
The flag was now changed (now the cmd, not the quirk), so this seems good now. Still, it's hard to follow as there are multiple cases to consider:
  - new-drv-old-fw: FW stops DMA early (1step), this DAI_CONFIG is ignored
  - old-drv-new-fw: FW stops DMA early as SW didn't send the quirk (1step), no DAI_CONFIG sent
  - new-drv-new-fw: FW delays DMA stop, DAI_CONFIG is sent and acted on (the actual DMA stop)

So if we wouldn't have to consider old FW, no new flags would be needed.